bin/node/runtime/src/lib.rs

```diff
@@ -441,6 +441,13 @@ parameter_types! {
 	pub const MaxIterations: u32 = 10;
 	// 0.05%. The higher the value, the more strict solution acceptance becomes.
 	pub MinSolutionScoreBump: Perbill = Perbill::from_rational_approximation(5u32, 10_000);
+	// The unsigned solution is operational, so as long as the average on_initialize weights are
+	// less than `MaximumBlockWeight * (1 - T::AvailableRatio)`, it can consume at most most this
+	// amount.
+	pub OffchainSolutionWeightLimit: Weight =
+		MaximumBlockWeight::get()
+			.saturating_sub(BlockExecutionWeight::get())
```

Comment:
This is supposed to be the "largest" extrinsic possible, but doesnt take into account weight introduced by on_initialize. Is that an issue?

Reply:
It seems you may be better off querying the current block weight as registered in System and using that to determine how much space you can fill?

Reply:
Indeed the intention is to be the big boy here, generate the largest possible extrinsic. Because this is now operational, it is fine even if we have some `on_initialize` weight in the block, as long as `average_on_initialize < operational_size`.

> It seems you may be better off querying the current block weight as registered in System and using that to determine how much space you can fill?

maybe... But there are flaws to this as well:

1. Configurability of the current system is better. A chain might want to allow simply 10% of block weight for npos solution, not always all. 
2. The transaction is almost likely produced with the data of block `N` and validated later for authoring against `N + 1` or `N + 2 ` or.. so this is a good _hint_, but not accurate.